packages/backend/src/ee/models/AiAgentModel.ts

```diff
@@ -148,4 +151,136 @@ export class AiAgentModel {
+                            organization_uuid: agent.organization_uuid,
+                            slack_channel_id: integration.channelId,
+                        });
+
+                        insertedIntegrations.push({
+                            type: integration.type,
+                            channelId: integration.channelId,
+                        });
+                    }
+                },
+            );
+
+            // Wait for all integration creation operations to complete
+            await Promise.all(integrationPromises);
+        }
+
+        return {
+            uuid: agent.ai_agent_uuid,
+            name: agent.name,
+            projectUuid: agent.project_uuid,
+            organizationUuid: agent.organization_uuid,
+            tags: agent.tags,
+            integrations: insertedIntegrations,
+        };
+    }
+
+    async updateAgent(
+        args: Pick<
+            ApiUpdateAiAgent,
+            'name' | 'projectUuid' | 'tags' | 'integrations'
+        > & {
+            agentUuid: string;
+            organizationUuid: string;
+        },
+    ): Promise<AiAgent> {
+        const [agent] = await this.database(AiAgentTableName)
+            .where('ai_agent_uuid', args.agentUuid)
+            .update({
+                name: args.name,
+                project_uuid: args.projectUuid,
+                tags: args.tags,
+            })
+            .returning('*');
+
+        if (agent.organization_uuid !== args.organizationUuid) {
+            throw new ForbiddenError('Organization not found');
+        }
+
+        const integrations = args.integrations || [];
+        const insertedIntegrations: ApiCreateAiAgent['integrations'] = [];
+
+        // Reset all integrations
+        await this.database(AiAgentIntegrationTableName)
+            .where('ai_agent_uuid', args.agentUuid)
+            .delete();
+
+        const integrationPromises = integrations.map(async (integration) => {
+            if (integration.type === 'slack') {
+                const [baseIntegration] = await this.database(
+                    AiAgentIntegrationTableName,
+                )
+                    .insert({
+                        ai_agent_uuid: agent.ai_agent_uuid,
+                        integration_type: integration.type,
+                    })
+                    .returning('*');
+
+                await this.database(AiAgentSlackIntegrationTableName).insert({
+                    ai_agent_integration_uuid:
+                        baseIntegration.ai_agent_integration_uuid,
+                    organization_uuid: agent.organization_uuid,
+                    slack_channel_id: integration.channelId,
+                });
+
+                insertedIntegrations.push({
+                    type: integration.type,
+                    channelId: integration.channelId,
+                });
+            }
+        });
+
+        await Promise.all(integrationPromises);
```

Comment:
all this operations should be in a transaction

Reply:
done 👍